app/controllers/orders/detail.js

```diff
@@ -236,19 +236,24 @@ export default GoodcityController.extend({
           this.send("promptCloseOrderModel", order, actionName);
           break;
         case "finish_processing":
-          if (this.get("processingChecklist").checklistCompleted(order)) {
-            this.send("changeOrderState", order, actionName);
-          } else {
-            this.genericAlertPopUp(
-              "order_details.logistics.checklist_incomplete"
-            );
-          }
+          this.send("verifyChecklistAndChangeState", order, actionName);
+          break;
+        case "start_dispatching":
+          this.send("verifyChecklistAndChangeState", order, actionName);
```

Comment:
case `finish_processing` and `start_dispatching` have same blocks, so they can be combined here.